src/app/shell/admin-tools/data/admins/admins.component.ts

```diff
@@ -103,26 +105,41 @@ export class AdminsComponent implements OnInit, OnDestroy {
     });
   }
 
-  /**
-   * This method block Admin By Id
-   */
-  onBlock(admin: UsersTable): void {
+  private openBlockModal(admin: BlockData): void {
     const dialogRef = this.matDialog.open(ConfirmationModalWindowComponent, {
       width: Constants.MODAL_SMALL,
       data: {
-        type: ModalConfirmationType.blockMinistryAdmin,
-        property: admin.pib,
+        type: admin.isBlocked ? ModalConfirmationType.blockMinistryAdmin : ModalConfirmationType.unBlockMinistryAdmin,
```

Comment:
again check? you call it from different methods, so you already know is it block or unblock
count how many times you do this check